### PR TITLE
Replace MurmurHash with Rust's DefaultHasher for CSS identifiers

### PR DESCRIPTION
**Pull request by jantimon**
Replace our custom MurmurHash2 implementation with Rust's built-in `DefaultHasher` (SipHash-1-3) for generating CSS identifiers

Key Changes:
- Replace `murmur_hash.rs` with `css_hash.rs` using Rust's `DefaultHasher`
- Generate base62 instead of base36 hashes to allow more characters in identifiers
- Always start generated identifiers with a letter to ensure CSS validity 
- Update hash length for better compatibility and readability

For example, hashes now look like:
```diff
- y_17k2ec6  // old format (base36)
+ yupb5bt    // new format (base62, starts with letter)
```

This change prepares us for implementing unique class name generation once Next.js adds support for global styles inside PostCSS CSS module files

### CODE REVIEW

File: packages/yak-swc/yak_swc/src/utils/css_hash.rs [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+  input.hash(&mut hasher);
+  let hash = hasher.finish();
+
+  // Convert to u32 to maintain similar output length as original
+  to_css_identifier(hash)
+}
+
+// Helper function to convert u32 to a valid CSS identifier
+fn to_css_identifier(mut num: u64) -> String {
+  const DIGITS: &[u8] = b"abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789_-";
```
Comment (Tyderion) [inline packages/yak-swc/yak_swc/src/utils/css_hash.rs:18]: maybe
```rs
const DIGITS: &[u8] = b"abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789_-";
const FIRST_LETTER_DIGITS: &[u8] = b"abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ";
``

und dann
```rs
  result.push(DIGITS[(num % FIRST_LETTER_DIGITS.len()) as usize]);
  num /= FIRST_LETTER_DIGITS.len();
...
    result.push(DIGITS[(num % DIGITS.len()) as usize]);
    num /= DIGITS.len();

```

File: packages/yak-swc/yak_swc/src/utils/css_hash.rs [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+  fn test_hash_length() {
+    let hash = hash_to_css("any string");
+    assert_eq!(hash.len(), 6);
+  }
+
+  #[test]
+  fn test_hash_is_base36() {
+    let hash = hash_to_css("base36 test");
+    assert!(hash.chars().all(|c| c.is_digit(36)));
+  }
```
[the remark this answers is not in the document]
Comment (jantimon) [reply packages/yak-swc/yak_swc/src/utils/css_hash.rs]: correct - replaced with a new test

File: packages/yak-swc/yak_swc/src/utils/css_hash.rs [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+  let mut hasher = DefaultHasher::new();
+  input.hash(&mut hasher);
+  let hash = hasher.finish();
+  to_css_identifier(hash)
+}
+
+// Helper function to convert u64 to a valid CSS identifier
+fn to_css_identifier(mut num: u64) -> String {
+  // CSS Identifiers must start with a letter
+  const FIRST_LETTER_DIGITS: &[u8] = b"abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ";
```
Comment (Mad-Kat) [inline packages/yak-swc/yak_swc/src/utils/css_hash.rs:17]: Shouldn't constants be moved to the module, so they don't get recreated every time this function runs?

File: packages/yak-swc/yak_swc/src/utils/css_hash.rs [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+  let first_letter_count = FIRST_LETTER_DIGITS.len() as u64;
+  // CSS Identifiers can contain letters, digits, hyphens, and underscores
+  const DIGITS: &[u8] = b"abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789";
+  let digit_count = DIGITS.len() as u64;
+
+  let mut result = Vec::new();
```
Comment (Mad-Kat) [inline packages/yak-swc/yak_swc/src/utils/css_hash.rs:23]:
```suggestion
  let mut result = Vec::with_capacity(6);
```

And maybe add another constant for the 6 char length?

**Review by Mad-Kat — APPROVED** [review]

File: packages/yak-swc/yak_swc/src/utils/css_hash.rs [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+  let hash = hasher.finish();
+  to_css_identifier(hash)
+}
+
+// Helper function to convert u64 to a valid CSS identifier
+fn to_css_identifier(mut num: u64) -> String {
+  // CSS Identifiers must start with a letter
+  const FIRST_LETTER_DIGITS: &[u8] = b"abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ";
+  let first_letter_count = FIRST_LETTER_DIGITS.len() as u64;
+  // CSS Identifiers can contain letters, digits, hyphens, and underscores
```
Comment (Tyderion) [inline packages/yak-swc/yak_swc/src/utils/css_hash.rs:19]: hyphens, and underscores <- not part of the selections

Comment (jantimon) [reply packages/yak-swc/yak_swc/src/utils/css_hash.rs:19]: yes that's correct - the class names and variables are a little bit better readable in dev mode without dashes and underscores

**Pull request merged**